app/code/Magento/Bundle/Test/Mftf/Test/StorefrontBundleCheckBoxOptionValidationTest.xml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontBundleCheckBoxOptionValidationTest">
+        <annotations>
+            <features value="Bundle"/>
+            <stories value="Bundle product validation before add to cart"/>
+            <title value="Customer should be able to see only one validation message for checkbox option group"/>
+            <description value="Customer should be able to see only one validation message for checkbox option group"/>
+            <severity value="MINOR"/>
+            <group value="Bundle"/>
+        </annotations>
+        <before>
+            <createData entity="ApiProductWithDescription" stepKey="simpleProduct1" before="bundleProduct"/>
+            <createData entity="ApiProductWithDescription" stepKey="simpleProduct2" after="simpleProduct1"/>
+            <createData entity="ApiBundleProduct" stepKey="bundleProduct"/>
+            <createData entity="CheckboxOption" stepKey="checkboxBundleOption">
+                <requiredEntity createDataKey="bundleProduct"/>
+            </createData>
+            <createData entity="ApiBundleLink" stepKey="createBundleLink1">
+                <requiredEntity createDataKey="bundleProduct"/>
+                <requiredEntity createDataKey="checkboxBundleOption"/>
+                <requiredEntity createDataKey="simpleProduct1"/>
+                <field key="qty">2</field>
+            </createData>
+            <createData entity="ApiBundleLink" stepKey="createBundleLink2">
+                <requiredEntity createDataKey="bundleProduct"/>
+                <requiredEntity createDataKey="checkboxBundleOption"/>
+                <requiredEntity createDataKey="simpleProduct2"/>
+                <field key="qty">4</field>
+            </createData>
+            <magentoCLI command="indexer:reindex" arguments="cataloginventory_stock" stepKey="reindex"/>
```

Comment:
Please, do not run the indexer directly, use the corresponding cron command

https://github.com/magento/magento-functional-tests-migration/pull/782/files#diff-652fb193d39d942beac21346810bf922R34